build error

can't  build latest git master (734f80a5d18821f4f6d0488996655c6e35ce160c).
boost version 1.58
    ...
    CXX      wallet/libbitcoin_wallet_a-rpcwallet.o
    In file included from wallet/rpcwallet.cpp:7:
    In file included from ./base58.h:21:
    In file included from ./script/standard.h:12:
    In file included from /opt/local/include/boost/variant.hpp:22:
    /opt/local/include/boost/variant/get.hpp:212:42: error: implicit instantiation of undefined template 'boost::STATIC_ASSERTION_FAILURE<false>'
        typedef ::boost::static_assert_test< sizeof(::boost::STATIC_ASSERTION_FAILURE< (((boost::detail::variant::holds_element<boost::variant< T0 , T1 , T2 , T3 , T4 , T5 , T6 , T7 , T8 , T9 , T10 , T11 , T12 , T13 , T14 , T15 , T16 , T17 , T18 , T19 >, U >::value)) == 0 ? false : true) >)> boost_static_assert_typedef_216 ;
                                            ^
    /opt/local/include/boost/variant/get.hpp:284:12: note: in instantiation of function template specialization 'boost::strict_get<const CScriptID &, CNoDestination, CKeyID, CScriptID, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>' requested here
        return strict_get<U>(operand);
            ^
    wallet/rpcwallet.cpp:2333:48: note: in instantiation of function template specialization 'boost::get<const CScriptID &, CNoDestination, CKeyID, CScriptID, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_>' requested here
                    const CScriptID& hash = boost::get<const CScriptID&>(address);
                                                ^
    /opt/local/include/boost/static_assert.hpp:87:26: note: template is declared here
    template <bool x> struct STATIC_ASSERTION_FAILURE;
                            ^
    1 error generated.
    make[2]: *** [wallet/libbitcoin_wallet_a-rpcwallet.o] Error 1
    make[1]: *** [all-recursive] Error 1
    make: *** [all-recursive] Error 1

--- a/src/wallet/rpcwallet.cpp
+++ b/src/wallet/rpcwallet.cpp
@@ -2330,7 +2330,7 @@ Value listunspent(const Array& params, bool fHelp)
         if (pk.IsPayToScriptHash()) {
             CTxDestination address;
             if (ExtractDestination(pk, address)) {
-                const CScriptID& hash = boost::get<const CScriptID&>(address);
+                const CScriptID& hash = boost::get<CScriptID>(address);
                 CScript redeemScript;
                 if (pwalletMain->GetCScript(hash, redeemScript))
                     entry.push_back(Pair("redeemScript", HexStr(redeemScript.begin(), redeemScript.end())));

